server/routes/handlers/getMatchingRecordRouteHandler.js

```diff
@@ -0,0 +1,67 @@
+const { settings } = require('../../config')
+const { getPaginationObject } = require('../helpers')
+const getMatchingRecordRouteHandler = (getMatchDetails, getCaseAndTemplateValues) => async (req, res) => {
+  const {
+    params: { courtCode, caseId, hearingId, defendantId },
+    session,
+    path,
+    query: queryParams
+  } = req
+
+  const templateValues = await getCaseAndTemplateValues(req)
+  templateValues.title = 'Review possible NDelius records'
+
+  const { data: { defendantName } } = templateValues
+
+  const response = await getMatchDetails(defendantId)
+
+  const { offenderMatchDetails } = response
+
+  if (!Array.isArray(offenderMatchDetails) || offenderMatchDetails.length === 0) {
+    return res.render('error', {
+      status: response.status || 500
+    })
+  }
+
+  // Parse page number and ensure it's a positive integer
```

Comment:
Not sure how helpful these comments are, and there are a lot in this function - can we make the code more readable if we think the comments are necessary?

Reply:
removed comments